### ISSUE TITLE
Blank screen on Ubuntu

### ISSUE THREAD

**Issue by larspensjo** [ethereum/meteor-dapp-wallet#78]
When I start the app, I get a blank screen.

Version: 0.3.5 64-bit linux
OS: Ubuntu 15.10 (almost fresh install).
Window manager: Unity

I first start geth manually, to verify that it runs and syncs correctly.
The Wallet app first shows the splash screen "Starting application..." for 1 second, then it is replaced by the "Ethereum wallet". The menus work, but the screen remains blank (white). If I resize it, it usually turns black. There are widgets on the screen, as I can see the cursor type change when hovering above them. I do not use "Additional drivers". That is, I don't have fglrx installed.

![screenshot from 2015-11-16 12-38-26](https://cloud.githubusercontent.com/assets/2412342/11181024/62cbe154-8c5f-11e5-9929-dde4deec9754.png)

The log output from the application looks ok:
```
CONNECT to IPC PATH: /home/lars/.ethereum/geth.ipc
Geth connection FOUND
Checking node sync status...
Time between last block 21s
No sync necessary, starting app!
Loading Interface at file:///lars/Downloads/Ethereum-Wallet-linux-x64-0-3-5/resources/app.asar/interface/wallet/index.html
CONNECT to IPC PATH: /home/lars/.ethereum/geth.ipc
IPCSOCKET 8 RESPONSE { jsonrpc: '2.0', method: 'eth_syncing', params: [], id: 8 } { id: 8, jsonrpc: '2.0', result: false } 


IPCSOCKET 8 RESPONSE { jsonrpc: '2.0', method: 'eth_accounts', params: [], id: 9 } { id: 9,
  jsonrpc: '2.0',
  result: 
```

**Comment by frozeman** [comment]
Could you please open the dev tools (Menu -> Develop -> Toogle…) and show me whats said?




Liebe GrüßeFabian

**Comment by larspensjo** [comment]
When I did toggle "Wallet UI", still nothing showed.

Next, I installed meteor downloaded the current master branch from github. After that, it works. That is, the separately downloaded app works, as well as hosting it myself using meteor + Chrome browser.

So I think there was something downloaded with meteor that made the difference.

**Comment by frozeman** [comment]
Im not sure, the wallet should work on ubuntu when you run the binary. Or did you ran the cloned one from github? (Yes then you need to run meteor and the wallet)

**Comment by petre108** [comment]
I am getting the same blank screen when running the binary on Ubuntu 14.04.3 LTS:

![blank screen](https://cloud.githubusercontent.com/assets/321786/11211615/593f6f90-8d39-11e5-9a0d-eebf59468685.png)

When I try to drag the white background I can see this:
![when dragging the white background](https://cloud.githubusercontent.com/assets/321786/11211752/505e5dae-8d3a-11e5-9e0d-a40d1cdba05d.png)

**Comment by frozeman** [comment]
Looks funky.
Could your try with the mist preview version: https://github.com/ethereum/mist/releases/tag/0.3.6

**Comment by petre108** [comment]
I get the same thing with with mist:
![mist blank](https://cloud.githubusercontent.com/assets/321786/11218446/be03fc88-8d5d-11e5-900a-dd683babcd3d.png)
oh and if I resize the window, it turns from white to black

**Comment by frozeman** [comment]
Could you try to run the develop version from the repo, and update your electron version to 0.35.0?
use the `wallet` branch and follow the instructions in the readme.

**Comment by petre108** [comment]
I updated electron, I installed mist from the wallet branch, then when I do
-------------------------------------------
:~/mist$ cd interface && meteor
[[[[[ ~/mist/interface ]]]]]                  

=> Started proxy.                             
=> Started MongoDB.                           
=> Started your app.                          

=> App running at: http://localhost:3000/
electron ./

-------------------------------------------------

I get nothing, Mist doesn't start.

**Comment by frozeman** [comment]
You start the interface in on terminal and in a second you run `electron .` from inside the mist folder again

**Comment by petre108** [comment]
got it,

it opens mist but with the same blank white screen.

Here is the output:

electron ./
CONNECT to IPC PATH: /home/123/.ethereum/geth.ipc
Node type:  geth
Network:  main
Start node from /home/123/mist/nodes/geth/linux-x64/geth
Stopping nodes...
Starting geth node...
Saved standard node "geth" to file: /home/123/.config/chromium/Mist/node
Saved network type "main" to file: /home/123/.config/chromium/Mist/network
Geth connection FOUND
Checking node sync status...
Time between last block 54598s
Sync finished, starting app!
Loading Interface at http://localhost:3000
CONNECT to IPC PATH: /home/123/.ethereum/geth.ipc

**Comment by frozeman** [comment]
you also need to start the wallet in another terminal on --port 3050

**Comment by petre108** [comment]
so what I did is
inside the mist folder:

1. changed  global.mode = 'mist'; inside the main.js file
2. cd interface && meteor
3. electron ./ (opens Mist, white screen as always)
4. changed  global.mode = 'wallet'; inside the main.js file
5. cd myapp/ && meteor --port 3050
6. electron ./ (opens myapp wallet, with the same white screen)

**Comment by frozeman** [comment]
3 needs to be run from the mist folder (`cd ../`).

can you open the console (ctrl/cmd + alt + i) and see whats there?

**Comment by petre108** [comment]
yes, 3 was run from inside the mist folder

here is how opening the console looks like in both Mist and the wallet:
![console open](https://cloud.githubusercontent.com/assets/321786/11271186/140bed46-8ecd-11e5-9e4d-e4ee6f522478.png)

**Comment by frozeman** [comment]
Can you click on develop -> toggle ... console ?

**Comment by petre108** [comment]
same result:
![toggle console](https://cloud.githubusercontent.com/assets/321786/11271872/91133bec-8ed1-11e5-95ce-ea64523e0e57.png)

**Comment by frozeman** [comment]
Could you click on the "Mist UI Wallet" ? i would like to see what the console says

**Comment by petre108** [comment]
here you go:
![html](https://cloud.githubusercontent.com/assets/321786/11272333/bf80ad9a-8ed4-11e5-8fc4-f0437b38cbba.png)
:

**Comment by frozeman** [comment]
so it seems to load the app, but doesn't show it.
Could you click on console on the top, so i see the console log?

**Comment by petre108** [comment]
console:
![console](https://cloud.githubusercontent.com/assets/321786/11272685/67a2c862-8ed7-11e5-8eec-bfc8d86e5595.png)

**Comment by frozeman** [comment]
So technically everything seems fine. Can you update electron on your ubuntu?
`npm update -g electron-prebuilt`

**Comment by petre108** [comment]
yes, it was already updated. same blank screen.
Don't worry about it too much, hopefully this is something very niche related to my specific ubuntu machine and all other users will be fine.

**Comment by frozeman** [comment]
You need to run electron 0.35.0 check with `electron --version`

**Comment by petre108** [comment]
$ electron --version
v0.35.0

**Comment by frozeman** [comment]
hm, then I'm not sure whats the problem. Can you simple run `electron` and see if it opens the default application?
Maybe it has to do with the enabled webgl. Could you look at main.js line 232 and 258 and set it to false and run again? Also try the other window options there if one fixes it.

**Comment by petre108** [comment]
electron seems to work fine:
![electron](https://cloud.githubusercontent.com/assets/321786/11273231/c4b98a9c-8eda-11e5-9b72-d06b2cbdaad7.png)

Also setting webgl to false on those two lines did not help. same blank screen.

**Comment by diglos** [comment]
Hi,

got the same problem but I may have a hint about this. I have a hybrid gpu laptop (Intel/nvidia). This only happens with the intel graphics card. When I switch to Nvidia everything looks fine. Could be a DRM issue?

**Comment by petre108** [comment]
I have an integrated AMD Radeon HD 6620G

**Comment by frozeman** [comment]
Works on my ubuntu. It seems to be a weird linux issue.

Probably related to:
https://github.com/atom/electron/issues/3878
https://github.com/atom/electron/issues/1226

**Comment by vmn551** [comment]
I wanted to chime in that I am having this same exact issue and am also using AMD Radeon HD 6620G

**Comment by frozeman** [comment]
This is an duplicate of https://github.com/ethereum/meteor-dapp-wallet/issues/78#issuecomment-175361031
Please try, as described there. Start electron using:
`electron . --ignore-gpu-blacklist`

**Comment by petre108** [comment]
Running 
./Ethereum-Wallet --ignore-gpu-blacklist
worked at last! :)

Thank you!

**Comment by gcmartinelli** [comment]
Worked for me as well (running Ubuntu on a Intel/Nvidia setup)

**Comment by gcmartinelli** [comment]
Tried this but program crashes.

Output:

`$ ./Ethereum-Wallet --ignore-gpu-blacklist
CONNECT to IPC PATH: /home/thx/.ethereum/geth.ipc
Geth connection FOUND
Checking node sync status...
Time between last block 6s
No sync necessary, starting app!
Loading Interface at file:///home/thx/Desktop/Ethereum-Wallet-linux-x64-0-3-8/resources/app.asar/interface/wallet/index.html
CONNECT to IPC PATH: /home/thx/.ethereum/geth.ipc
Segmentation fault (core dumped)`

**Comment by frozeman** [comment]
Thats exactly why these GPUs are blacklisted... because they can cause crashes. So i don't know if its smart to append this flag by default now. It only happens on some custom linux systems.

**Comment by gcmartinelli** [comment]
Running Ubuntu 15.10 with Intel/Nvidia (geforce 740m) cards

**Comment by frozeman** [comment]
I don't really know a solution for this, can you google for "electron gpu blacklist linux", if you find some more?

**Comment by taoteh1221** [comment]
Can confirm  "--ignore-gpu-blacklist" fixed the issue for my SAPPHIRE DUAL-X 100373L Radeon R9 280 3GB. Used the wallet for an hour or so setting up a live multisig wallet, never had any crashes.

**Comment by taoteh1221** [comment]
Sorry, forgot to mention: Above comment was on Linux Mint 17.2, the wallet works FINE on the same machine booted into Win10.

**Comment by taoteh1221** [comment]
Spoke too soon, just crashed (v0.3.8) on Win10, but only after trying to send a transaction from a mulltisig wallet. I guess this particular video card really is no good for Mist, even on Windows.

**Comment by frozeman** [comment]
I will release the next version with webgl deactivated, please try this one then.

**Comment by taoteh1221** [comment]
Ok, will do.

**Comment by frozeman** [comment]
Can you try: https://github.com/ethereum/mist/releases/tag/0.3.9

**Comment by taoteh1221** [comment]
So far, so good on Win10 with v0.3.9. --v0.3.8-- kept crashing immediately after blocks loaded and it said "Starting Application". The wallet would show for about half a second and then crash / close out. 

v0.3.9 is NOT crashing at all yet on Win10. I have closed / restarted it several times and it SEEMS to run very well. I will run further tests with transactions etc on Win10, AND I will boot into Linux Mint 17.2 and try v0.3.9 there as well. Probably won't get to it today, but I will report back my findings tomorrow...I am swamped with work from 4 web development customers today. 

Thanks for the great product support, and Mist is coming along nicely I must say, really awesome cryptocurrency app, best out there hands down in my opinion. You all do great work.

**Comment by taoteh1221** [comment]
Same blank screen in Linux Mint 17.2 with v0.3.9, BUT when I switched from the "xorg" open source Radeon video driver (the default) over to the proprietary "grflx" Radeon video driver released by the video card manufacturer, BOTH v0.3.8 and v0.3.9 worked fine on my machine WITHOUT using "--ignore-gpu-blacklist". 

I did notice on Linux Mint that v0.3.9 did not retrieve my multisig contract wallet I setup recently in v0.3.8, but when I switched back to v0.3.8 my mutisig wallet was still registered and showing.

I will have more time to test tomorrow, initiating transactions etc...that seems to be what caused my initial crash in Win10.

**Comment by frozeman** [comment]
@taoteh1221 if you start the wallet in dev mode following the instructions in the readme, your wallet has a different `location.origin` as its loaded from `localhost:3000`, so you won't see your wallets, as in the build version if has as origin the `file://` protocol

**Comment by gcmartinelli** [comment]
Also getting blank screen on Ubuntu 15.10, Nvidia/Intel graphics and Xorg driver (default on Ubuntu).
Works only with proprietary Nvidia drivers (which are not an option since they are very unstable).

**Comment by taoteh1221** [comment]
@frozeman I fully tested v0.3.9 on that Win10 / Radeon video machine today, sending transactions in / out of a multisig wallet from local accounts, seems to be solid now on my setup. No issues at all during about a 30 minute period of constant use. 

v0.3.9 on Linux Mint on same machine seems solid too with the "grflx" video driver, but even though I did NOT start the wallet in dev mode it did not show my saved contracts from v0.3.8. I re-imported them and everything seems to still work well.

**Comment by taoteh1221** [comment]
@gcmartinelli I think you may be able to force the wallet to NOT use Nvidia for the graphics per app? My Dad had an Nvidia/Intel graphics laptop once, and I think there was an app built into the drivers suite that let him pic which graphics processor was used per application? Not sure though. I do know my Acer V3112PC2P6 laptop ONLY has Intel graphics, and the wallet works fine on that machine with no tweaking needed.

**Comment by gcmartinelli** [comment]
@taoteh1221 I'll look around. I believe I know what you are talking about and it is already set to use the Intel card. The option is named "Prime Profiles" on the nvidia-settings app.

**Comment by knocte** [comment]
FYI same bug here in Ubuntu 14.04.1 LTS. The workaround `./Ethereum-Wallet --ignore-gpu-blacklist` works.

**Comment by frozeman** [comment]
Duplicate of https://github.com/ethereum/mist/issues/75, please continue there.

**Comment by nand2** [comment]
FYI same bug here, Ubuntu 14.04 with dual GPUs Intel/Nvidia, the workaround with --ignore-gpu-blacklist works.

**Comment by robertclarkson** [comment]
Same here Unbuntu 14.04 with Geforce GPU  --ignore-gpu-blacklist works

**Comment by ximo1984** [comment]
Same here Unbuntu 14.04 with Geforce GPU --ignore-gpu-blacklist works

**Comment by yanncarlier** [comment]
ethereumwallet and 'ethereumwallet --ignore-gpu-blacklist' gives me a blank screen on ubuntu 18.04
